app/controllers/v2/content_api_prototype/content_items/content_controller.rb

```diff
@@ -0,0 +1,109 @@
+module V2
+  module ContentApiPrototype
+    module ContentItems
+      class ContentTypeResolver
+        def initialize(content_type)
+          self.content_type = content_type
+        end
+
+        def resolve(object)
+          if object.is_a?(Hash)
+            resolve_hash(object)
+          elsif object.is_a?(Array)
+            resolve_array(object)
+          else
+            object
+          end
+        end
+
+      private
+
+        def resolve_hash(hash)
+          hash.inject({}) do |memo, (key, value)|
+            memo.merge(key => resolve(value))
+          end
+        end
+
+        def resolve_array(array)
+          if array.all? { |v| v.is_a?(Hash) }
+            array = array.map(&:symbolize_keys)
+            content_for_type = extract_content(array)
+          end
+
+          if content_for_type
+            content_for_type[:content]
+          else
+            array.map { |e| resolve(e) }
+          end
+        end
+
+        def extract_content(array)
+          array.detect { |h| h[:content_type] == content_type && h[:content] }
+        end
+
+        attr_accessor :content_type
+      end
+
+      class ContentController < ApplicationController
+        def by_base_path
+          edition = Edition
+            .where(base_path: base_path)
+            .order(user_facing_version: "DESC")
+            .first!
+
+          render json: present(edition)
+        end
+
+        def by_content_id
+          edition = Edition
+            .with_document
+            .find_by!(
+              documents: {
+                content_id: path_params[:content_id],
+                locale: path_params[:locale]
+              },
+              user_facing_version: path_params[:user_facing_version]
+            )
+
+          render json: present(edition)
+        end
+
+      private
+
+        def present(edition)
+          presented_edition = Presenters::EditionPresenter
+            .new(edition, draft: edition.draft?)
+            .for_content_store(0)
+
+          presented_edition = resolve_text_html(presented_edition)
+
+          inject_links(presented_edition, edition)
+        end
+
+        def resolve_text_html(presented_edition)
+          resolver = ContentTypeResolver.new("text/html")
+          presented_edition[:details] = resolver.resolve(presented_edition[:details])
+          presented_edition[:description] = resolver.resolve(presented_edition[:description])
+          presented_edition
+        end
+
+        def inject_links(presented_edition, edition)
+          change_history_presenter = Presenters::ChangeHistoryPresenter.new(edition)
+          change_notes = change_history_presenter.send(:change_notes)
+
+          presented_edition[:details][:change_history] = presented_edition[:details][:change_history].each_with_index.map do |presented_change_note, index|
```

Comment:
Thinking further about this I think you're right to have this as change notes. Maybe we should just have links to previous and next editions and allow you to paginate that way rather than having a full list?

Ideally it feels like there should be an endpoint that exposes a paginated list of the history but I don't think that fits into this prototype